Sources/Networking/HTTPClient.swift

```diff
@@ -160,62 +162,112 @@ private extension HTTPClient {
         self.start(request: request)
     }
 
-    func handle(urlResponse: URLResponse?,
-                request: Request,
-                urlRequest: URLRequest,
-                data: Data?,
-                error networkError: Error?) {
-        var statusCode: HTTPStatusCode = .networkConnectTimeoutError
-        var jsonObject: [String: Any]?
-        var httpResponse: HTTPResponse? = HTTPResponse(statusCode: statusCode, jsonObject: jsonObject)
-        var receivedJSONError: Error?
-
-        if networkError == nil {
-            if let httpURLResponse = urlResponse as? HTTPURLResponse {
-                statusCode = .init(rawValue: httpURLResponse.statusCode)
-                Logger.debug(Strings.network.api_request_completed(request.httpRequest, httpCode: statusCode))
-
-                if statusCode == .notModified || data == nil {
-                    jsonObject = [:]
-                } else if let data = data {
-                    do {
-                        jsonObject = try JSONSerialization.jsonObject(with: data) as? [String: Any]
-                    } catch let jsonError {
-                        Logger.error(Strings.network.parsing_json_error(error: jsonError))
-
-                        let dataAsString = String(data: data, encoding: .utf8) ?? ""
-                        Logger.error(Strings.network.json_data_received(dataString: dataAsString))
-
-                        receivedJSONError = jsonError
-                    }
+    // swiftlint:disable:next function_body_length
+    func parse(urlResponse: URLResponse?,
```

Comment:
These changes are hard to follow in the diff, but I did this very slowly thinking about every possible path. Also our test coverage gives me a lot of confidence these are equivalent.